src/librustc_lint/nonstandard_style.rs

```diff
@@ -201,20 +208,26 @@ impl NonSnakeCase {
             })
         }
 
+        let name = &ident.name.as_str();
+
         if !is_snake_case(name) {
             let sc = NonSnakeCase::to_snake_case(name);
-            let msg = if sc != name {
-                format!("{} `{}` should have a snake case name such as `{}`",
-                        sort,
-                        name,
-                        sc)
+
+            let msg = format!("{} `{}` should have a snake case name", sort, name);
+            let mut err = cx.struct_span_lint(NON_SNAKE_CASE, ident.span, &msg);
+
+            if !ident.span.is_dummy() {
```

Comment:
is this actually an issue? if so, why don't we have the same problem for `NON_CAMEL_CASE_TYPES`

Reply:
It's only an issue when we're linting `--crate-name NonSnakeCase`, because there's no span when we create the diagnostic from the command line.

Reply:
ah cool, can you add a comment explaining just that?

Reply:
Done.